src/components/structures/UserSettings.js

```diff
@@ -379,19 +395,42 @@ module.exports = React.createClass({
                             Disable inline URL previews by default
                         </label>
                     </div>
+                    { settingsLabels.map( setting => {
+                        return <div className="mx_UserSettings_toggle" key={ setting.id }>
+                            <input id={ setting.id }
+                                   type="checkbox"
+                                   defaultChecked={ syncedSettings[setting.id] }
+                                   onChange={ e => UserSettingsStore.setSyncedSetting(setting.id, e.target.checked) }
+                            />
+                            <label htmlFor={ setting.id }>
+                                { setting.label }
+                            </label>
+                        </div>
+                    })}
+                    { themes.map( setting => {
+                        return <div className="mx_UserSettings_toggle" key={ setting.id + "_" + setting.value }>
```

Comment:
can't you make this a named function rather than an arrow function in the middle of the jsx? this is pretty unreadable.